server/controllers/googleApiController.js

```diff
@@ -66,44 +89,38 @@ function authenticateUser(req, res) {
     .catch((err) => res.status(422).json(err));
 }
 
-function getAvailability(req, res) {
+async function getAvailability(req, res) {
   const { query } = req;
   const date = `${query.year}/${query.month}/${query.date}`; // 2020-08-20
   const meetingLength = parseInt(query.meetingLength); // "30min" => 30
-  const { availableFrom, availableTo, timeZone } = query;
+  const { timeZone, calendarUrl } = query;
+  let user;
 
-  //create time availability range based on user availability preference
-  const availabilityStart = moment(`${date} ${availableFrom}`);
-  const availabilityEnd = moment(`${date} ${availableTo}`);
-  const range = moment.range(availabilityStart, availabilityEnd);
-
-  //slice the range by meetinglength into different slots
-  let timeSlot = Array.from(range.by("minutes", { step: meetingLength }));
-  timeSlot.pop();
+  findUserByUrl(calendarUrl)
+    .then((dbModel) => {
+      user = dbModel;
+      user.availabilityStart = moment(`${date} ${user.availableHoursFrom}`);
+      user.availabilityEnd = moment(`${date} ${user.availableHoursTo}`);
+      // ===================================
+      //For now, hard code the refresh_token
+      return getGoogleCalendarApi(oAuth2Client, {
+        refresh_token:
```

Comment:
https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/#Token-types

make sure the refresh token is not being overused. as mentioned, if possible we should be using access token until it has expired, and then use the refresh token to get a new access token

Reply:
thanks for the link, very helpful

Reply:
resolved by checking if the access token expired(line 124)
if it is expired, refresh the access token using OAuth library, update the user info in the database, and send the updated user back (line 45, line 125)
code readability is not very high as there are then chains called inside chains, might reorganize the codes later